src/Form/LegacyConsumer/resources/js/conditinal-fields.js

```diff
@@ -0,0 +1,222 @@
+			const {field, operator, value} = visibilityCondition;
+
+			const inputs = donationForm.querySelectorAll(`[name="${field}"]`);
+			let hasFieldController = !!inputs.length;
+
+			// Do not apply visibility conditions if field controller does not exit in DOM.
+			if (!hasFieldController) {
+				return;
+			}
+
+			inputs.forEach((input) => {
+				const fieldType = input.getAttribute('type');
+				const comparisonResult = compareWithOperator( operator, input.value, value );
+
+				if (fieldType && (fieldType === 'radio' || fieldType === 'checkbox')) {
+					if (input.checked && comparisonResult ) {
+						visible = true;
+					}
+				} else if ( comparisonResult ) {
+					visible = true;
+				}
+			});
+
+			// Show or Hide field wrapper.
+			visible ?
+				fieldWrapper.classList.remove('give-hidden') :
+				fieldWrapper.classList.add('give-hidden');
+		}
+	}
+
+	/**
+	 * Setup state for condition visibility settings.
+	 * state contains list of watched elements per donation form.
+	 *
+	 * @unreleased
+	 */
+	function addVisibilityConditionsToStateForDonationForm(donationForm) {
+		const uniqueDonationFormId = donationForm.getAttribute('data-id');
+		const watchedFields = getWatchedElementNames(donationForm);
+
+		// Add donation form to state only if visibility conditions exiting for at least form field.
+		if (!uniqueDonationFormId || !Object.entries(watchedFields).length) {
+			return false;
+		}
+
+		state[uniqueDonationFormId] = {
+			watchedElements: watchedFields,
+			...state[uniqueDonationFormId]
+		}
+	}
+
+	/**
+	 * Setup state for condition visibility settings.
+	 * state contains list of watched elements per donation form.
+	 *
+	 * @unreleased
+	 * @returns {Promise<void>}
+	 */
+	function addVisibilityConditionsToStateForAllDonationForm() {
+		document.querySelectorAll('form.give-form')
+			.forEach(function (donationForm) {
+				const uniqueDonationFormId = donationForm.getAttribute('data-id');
+				const watchedFields = getWatchedElementNames(donationForm);
+
+				// Add donation form to state only if visibility conditions exiting for at least form field.
+				if (!uniqueDonationFormId || !Object.entries(watchedFields).length) {
+					return false;
+				}
+
+				state[uniqueDonationFormId] = {
+					watchedElements: watchedFields,
+					...state[uniqueDonationFormId]
+				}
+			});
+	}
+
+	/**
+	 * @unreleased
+	 */
+	function applyVisibilityConditionsAttachedToWatchedField(donationForm, fieldName) {
+		const uniqueDonationFormId = donationForm.getAttribute('data-id');
+
+		// Exit if field is not element of donation form.
+		if (
+			!donationForm ||
+			!uniqueDonationFormId ||
+			!state.hasOwnProperty(uniqueDonationFormId)
+		) {
+			return false;
+		}
+
+		const formState = state[uniqueDonationFormId];
+		const watchedElementNames = Object.keys(formState.watchedElements);
+
+		// Exit if field is not in list of watched elements.
+		if (!watchedElementNames.includes(fieldName)) {
+			return false;
+		}
```

Comment:
No need to extract all keys:
```js
if ( !(fieldName in formState.watchedElements) ) {
  return;
}

Reply:
fixed: https://github.com/impress-org/givewp/compare/125dc5c0628c...4721eb02efa0